bootstore/src/schemes/v0/state_initial_member.rs

```diff
@@ -0,0 +1,411 @@
+                );
+
+                if let Some(Ok(ApiOutput::RackSecret(rack_secret))) =
+                    &output.api_output
+                {
+                    // We already know the rack secret so respond to the
+                    // peer.
+                    decrypt_and_send_share_response(
+                        from,
+                        request_id,
+                        &self.pkg,
+                        &mut self.distributed_shares,
+                        rack_secret,
+                    )
+                } else {
+                    let expiry = common.clock + common.config.learn_timeout;
+                    self.pending_learn_requests
+                        .insert(from, RequestMetadata { request_id, expiry });
+                    output
+                }
+            }
+        };
+
+        // This is a terminal state
+        (self.into(), output)
+    }
+
+    fn handle_response(
+        mut self,
+        common: &mut FsmCommonData,
+        from: Baseboard,
+        request_id: Uuid,
+        response: ResponseType,
+    ) -> (State, Output) {
+        use ResponseType::*;
+        let output = match response {
+            InitAck => {
+                if let Some(rack_init_state) = &mut self.rack_init_state {
+                    if rack_init_state.on_ack(from) {
+                        return (
+                            self.into(),
+                            ApiOutput::RackInitComplete.into(),
+                        );
+                    }
+                }
+                Output::none()
+            }
+            Share(share) => self.on_share(common, from, request_id, share),
+            Pkg(_) => ApiError::UnexpectedResponse {
+                from,
+                state: self.name(),
+                request_id,
+                msg: response.name(),
+            }
+            .into(),
+            Error(error) => ApiError::ErrorResponseReceived {
+                from,
+                state: self.name(),
+                request_id,
+                error,
+            }
+            .into(),
+        };
+
+        // This is a terminal state
+        (self.into(), output)
+    }
+
+    fn tick(mut self, common: &mut FsmCommonData) -> (State, Output) {
+        // Check for rack initialization timeout
+        if let Some(rack_init_state) = &mut self.rack_init_state {
+            if rack_init_state
+                .timer_expired(common.clock, common.config.rack_init_timeout)
+            {
+                let unacked_peers = rack_init_state.unacked_peers();
+                *rack_init_state = RackInitState::Timeout;
+                return (
+                    self.into(),
+                    ApiError::RackInitTimeout { unacked_peers }.into(),
+                );
+            }
+        }
+        self.pending_learn_requests
+            .retain(|_, metadata| metadata.expiry >= common.clock);
+
+        (self.into(), common.rack_secret_state.on_tick(common.clock))
```

Comment:
If `common` needs ticking, it seems weird that it's done by the states instead of by the caller (since this means that each state implementation has to remember to call it).

Reply:
No longer relevant as of [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98)